common/serde-reflection/src/de.rs

```diff
@@ -308,6 +311,18 @@ impl<'de, 'a> de::Deserializer<'de> for Deserializer<'de, 'a> {
     {
         self.format.unify(Format::TypeName(name.into()))?;
         let mut formats = vec![Format::Unknown; len];
+        if self.tracer.config.record_values_for_tuple_structs {
+            // If a value was recorded during serialization, use it.
+            if let Some((format, hint)) = self.tracer.get_recorded_value(self.records, name) {
+                let result = || visitor.visit_seq(hint.seq_values()?.into_seq_deserializer());
```

Comment:
What's your rationale for doing this? why not just explicitly call .map_err on this instead of wrapping it in a closure?

Reply:
The lambda let me use a question mark after `hint.seq_values()`. The alternatives I could think of seemed worse (as in very verbose) but maybe I missed something.